src/plugins/data_source_management/public/components/utils.ts

```diff
@@ -282,11 +284,13 @@ export const handleDataSourceFetchError = (
 ) => {
   changeState({ showError: true });
   if (callback) callback([]);
-  notifications.addWarning(
-    i18n.translate('dataSource.fetchDataSourceError', {
-      defaultMessage: 'Failed to fetch data source',
-    })
-  );
+  notifications.add({
+    title: i18n.translate('dataSource.fetchDataSourceError', {
+      defaultMessage: 'Failed to fetch data sources',
+    }),
+    text: toMountPoint(getReloadButton()),
```

Comment:
perhaps we can show the error in more detail similar to how the Discover page shows the error response if clicked in the toast.

Reply:
This current implementation based on requirement from UX. Since this is more of network call failure, user needs to reload the page in order to trigger another call.